### ISSUE TITLE
Using different stoploss for long and short trades

### ISSUE THREAD

**Issue by moinonin** [freqtrade/freqtrade#7733]
## Describe your environment

  * Operating system: Linux mint Una
  * Python Version: 3.10.4
  * CCXT version: 1.91.55
  * Freqtrade Version: freqtrade docker-9df10c6b
  
## Your question
I am trying to combine two strategies to use in the futures market. I have run hyperopt on combined strategy and there seems not to be convergence. Does any one knows how I can use different stoploss for short and long trades? Comments highly appreciated.
*Ask the question you have not been able to find an answer in the [Documentation](https://www.freqtrade.io/en/latest/)*

**Comment by xmatthias** [comment]
You can simply implement the [custom stoploss](https://www.freqtrade.io/en/stable/strategy-callbacks/#custom-stoploss) callback - and return whatever you like (you can get the long/short question through `trade.is_short`.

Assuming it should not be trailing - you'll need to use one of the helpers - examples are in the docs.

**Comment by italodamato** [comment]
I have a similar problem: I want to use a custom stoploss value at every trade. For long trade it's easy: just set stoploss to a very low value like -0.9 and then the callback adjusts it.
The problem for short trades is that the stoploss ratio doesn't have an upper bound (like -1 for long trades). Meaning that if the custom stoploss value is actually higher than the value coming from the -.9 calculation, the value gets ignored since the stoploss can just get tighter by design.

Is there a way to prevent this?

**Comment by xmatthias** [comment]
-0.9 will always mean 90% loss - the direction of the trade is irrelevant (it's a Relative value).

So I'm not sure i understand what you mean exactly.

**Comment by italodamato** [comment]
Yes, that makes sense - I just got confused :)

**Comment by moinonin** [comment]
This is to confirm that using different stoploss worked as @xmatthias suggested. This is what I did:
```
    minimal_roi = {
        "0": 0.09,
        "255": 0.064,
        "578": 0.028,
        "705": 0
    }

    # Stoploss:
    stoploss = -0.25 # 0.10
```
Then after setting trailing stoplosses, I added the logic to get long trades and set a much smaller stoploss as follows:
-----------
```
        if current_profit > 0.05:
                return stoploss_from_open(0.03, current_profit, is_short=trade.is_short)
        if current_profit > 0.035:
                return stoploss_from_open(0.015, current_profit, is_short=trade.is_short)
        if trade.is_short == False:
            return -0.026
```
-----------
This is helpful because you can run hyperopt differently for long and short trades and later combine in one strategy. Possibly this could be added to the documentation.

**Comment by lekinvas49** [comment]
Hello Sorry,

I tried to implement custom_stoploss callback to make the difference between short and long, using trade.is_short but I keep having the same error while running dry run and backtest of my strategy.

Here was my code:

#Optimal stoploss designed for the strategy.
#This attribute will be overridden if the config file contains "stoploss".
    stoploss = -0.75 

    #custom stoploss for short 
    use_custom_stoploss=True
    def custom_stoploss(self, pair: str, trade: 'Trade', current_time: datetime,
                        current_rate: float, current_profit: float, **kwargs) -> float:
         if trade.is_short == True:
            return stoploss_from_open(-0.25, current_rate, is_short=trade.is_short)
         if trade.is_short == False:
            return -0.75
            
           ==> Any suggestions? Thank you guys!

**Comment by xmatthias** [comment]
stating "same error" without actually listing the error (you didn't comment in this issue before) will not work out.
the code looks ok at first glance - but without knowing the error - we'll not be able to help you.
